### DISCUSSION TITLE
how to release formula transformations memory

### DISCUSSION THREAD

**Discussion by yiminghe** [handsontable/hyperformula#904] [Feature requests]
when operating table for a long time, all formula transformations will be saved inside lazilyTransformingAstService, how to eager apply and release unused transformation ?

**Comment by MartinDawson** [comment]
You would just have to create the methods yourself to clear `transformations` and `combinedTransformer`.

What's the use case for this though? As far as I can tell it looks to be needed for the calculations and for Undo/Redo functionality.

**Reply by yiminghe** [reply]
If operating table for a long time, memory keeps growing. undo redo will splice according to undoLimit, but transformations will not.
